### PR TITLE
Add derivations to build docker images for cli and delegate server.

### PR DESCRIPTION
**Pull request by nazrhom**
This allows us to make some progress on #180 as osx users can now run the cli and delegate server through docker (thus fixing the socket issue).

There is still the caveat that building this derivation requires access to a linux machine (i.e. remote nix builder)

### CODE REVIEW

**Comment by MangoIV** [comment]
where did my changes wrt cross go? are they on another branch that hasn't been merged yet?

**Comment by MangoIV** [comment]
also, do you want me to change it so it's only buildable on linux liked you asked internally?

**Comment by nazrhom** [comment]
@MangoIV I created another branch, the changes you did for cross-compilation should still be in the other one

**Comment by nazrhom** [comment]
> also, do you want me to change it so it's only buildable on linux liked you asked internally?

Yeah, please! I think it makes sense to only add it for the platforms on which it would work

File: scripts/spin-up-new-devnet.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh -e
+#!/bin/bash -e
```
Comment (MangoIV) [inline scripts/spin-up-new-devnet.sh]:
```suggestion
#!/usr/bin/env bash -e
```

File: scripts/stop-demo.sh [hunk 2]
```diff
@@ -2,3 +2,4 @@
 
 docker-compose down
 docker-compose --profile hydra-node down
+docker-compose --profile delegate-server down
```
Comment (uhbif19) [inline scripts/stop-demo.sh:3]: Should not it be the other way around? Like delegate server depends on hydra-node, so should be killed first.

And I still do not know what is wrong with `depends_on`, which seems to do the same thing automatically.   I remember there was some problem with it... Like it does not ensure container started by health-check.

File: docker-compose.yaml [hunk 3]
```diff
@@ -120,6 +120,51 @@ services:
         ipv4_address: 172.16.238.30
     restart: always
 
+  devnet-server-1:
+    image: hydra-auction-delegate:latest
```
Comment (uhbif19) [inline docker-compose.yaml:124]: So, I am missing:

* Where does building take place?
* Where do it stores images? Locally I presume?

Does our CI automatically call whose derivations building images?

Comment (uhbif19) [reply docker-compose.yaml:124]: Could you please explain how to call building locally.

File: scripts/stop-demo.sh (hunk 2 above)
Comment (nazrhom) [reply scripts/stop-demo.sh:3]: `depends_on` is a bit tricky, as it does not care that whatever is running inside the container is "ready". I.e. we have to manually wait for the socket to be present from cardano-node, before running anything else...

Comment (uhbif19) [reply scripts/stop-demo.sh:3]: Seems like now is `condition: service_healthy` for that. While I am not sure it actually works well.

Anyway, our current code does not seem to have any waits, so I works the same as plain depends_on.

**Review by uhbif19 — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -38,9 +38,16 @@ Will build the CLI application and link it to `./result/bin/hydra-auction`.
...
+which will start a single cardano-node,
+3 hydra-nodes for the delgates,
+and 3 instances of the delegate server.
+
+Before starting the demo we will need to build the delegate server docker images.
+This is done through nix and can be done by `make build-docker`.
+This command will build the image and load it as `hydra-auction-delegate:latest`.
 
 We recommend using the `spin-up-new-devnet.sh` script
+to run the demo and start the components,
```
Comment (uhbif19) [inline README.md:50]: I think "the demo" is the whole process, including what we do in frontend cli manually.

**Pull request merged**